### PR TITLE
Fix roots

### PR DESCRIPTION
**Pull request by aliang8**
Fixed the matchFn for nthRoot rules. Fixed build notation.

### CODE REVIEW

File: lib/rules.js [hunk 1]
```diff
@@ -195,13 +195,26 @@ export const ABSOLUTE_VALUE = defineRuleString('|-#a|', '#a')
...
+            if (isNthRoot && validRadicand) {
+                // get the fraction exponent, simplify it, and
+                // see if the new fraction = old fraction
+                const radicand = node.args[0]
+                const variable = radicand.args[0]
+                const exponent = radicand.args[1]
+                let index = node.args[1]
+                const oldRoot = build.div(exponent, index)
+                const newRoot = applyRule(SIMPLIFY_FRACTION,
+                                          build.div(exponent, index))
```
Comment (k4b7) [inline lib/rules.js]: What if the index or the exponent aren't numbers?

File: lib/rules.js [hunk 2]
```diff
@@ -195,13 +195,26 @@ export const ABSOLUTE_VALUE = defineRuleString('|-#a|', '#a')
...
-            validRadicand = query.isPow(node.args[0])
+            let isNthRoot = (node.op == 'nthRoot')
+            let validRadicand = query.isPow(node.args[0])
+            if (isNthRoot && validRadicand) {
+                // get the fraction exponent, simplify it, and
+                // see if the new fraction = old fraction
+                const radicand = node.args[0]
+                const variable = radicand.args[0]
+                const exponent = radicand.args[1]
+                let index = node.args[1]
```
Comment (k4b7) [inline lib/rules.js]: You could also do, 
```
if (query.isNthRoot(node)) {
  const [radicand, index] = node.args
  if (query.isPow(radicand)) {
    const [base, exponent] = radicand.args
    if (query.isNumber(exponent) && query.isNumber(index)) {
      ...
    }
  }
}
```

File: lib/rules.js [hunk 3]
```diff
@@ -257,28 +266,23 @@ export const EVALUATE_DISTRIBUTED_NTH_ROOT = defineRule(
...
         if (isDistributed) {
-            canEvaluate = node.args.every(arg => canApplyRule(CANCEL_EXPONENT, arg) || canApplyRule(NTH_ROOT_VALUE, arg))
+            canEvaluate = node.args.some(arg => canApplyRule(CANCEL_EXPONENT, arg) || canApplyRule(NTH_ROOT_VALUE, arg))
         }
         return (isDistributed && canEvaluate) ? {node} : null
     },
 
     (node) => {
-        const result = build.applyNode(
+        const result = build.apply(
```
Comment (k4b7) [inline lib/rules.js]: `const result = build.mul(...node.args.map(nthRoot => { ... }))`

File: lib/rules.js (hunk 2 above)
Comment (k4b7) [reply lib/rules.js]: I just published a new version of `math-nodes` that has `isNthRoot`.

File: package.json [hunk 4]
```diff
@@ -19,7 +19,7 @@
     "babel-plugin-transform-object-rest-spread": "^6.23.0",
     "babel-polyfill": "^6.23.0",
     "babel-preset-es2015": "^6.24.1",
-    "eslint": "^3.15.0",
+    "eslint": "^4.0.0",
```
Comment (k4b7) [inline package.json]: Why this change?

**Review by k4b7 — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (aliang8) [reply package.json]: Sorry, I did an autoupgrade of all my dependencies. Will fix!

File: package.json [hunk 5]
```diff
@@ -21,8 +21,8 @@
     "babel-preset-es2015": "^6.24.1",
     "eslint": "^3.15.0",
     "jest": "^20.0.4",
-    "pre-commit": "^1.1.3",
-    "webpack": "^2.4.1"
+    "pre-commit": "^1.2.2",
+    "webpack": "^2.6.1"
```
Comment (k4b7) [inline package.json:25]: These dev dependency changes aren't strictly necessary.  Not a huge deal, but please try to avoid updating unless necessary.

**Review by k4b7 — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (aliang8) [reply package.json:25]: Got it!

**Pull request merged**